pkg/types/config/openstack/openstack.go

```diff
@@ -0,0 +1,103 @@
+	FlavorName       string            `json:"tectonic_openstack_worker_flavor_name,omitempty" yaml:"flavorName,omitempty"`
+	ExtraSGIDs       []string          `json:"tectonic_openstack_worker_extra_sg_ids,omitempty" yaml:"extraSGIDs,omitempty"`
+	LoadBalancers    []string          `json:"tectonic_openstack_worker_load_balancers,omitempty" yaml:"loadBalancers,omitempty"`
+	WorkerRootVolume `json:",inline" yaml:"rootVolume,omitempty"`
+}
+
+// WorkerRootVolume converts worker rool volume related config.
+type WorkerRootVolume struct {
+	IOPS int    `json:"tectonic_openstack_worker_root_volume_iops,omitempty" yaml:"iops,omitempty"`
+	Size int    `json:"tectonic_openstack_worker_root_volume_size,omitempty" yaml:"size,omitempty"`
+	Type string `json:"tectonic_openstack_worker_root_volume_type,omitempty" yaml:"type,omitempty"`
+}
+
+// Credentials converts credentials related config.
+type Credentials struct {
+	AuthURL           string `json:"tectonic_openstack_credentials_auth_url,omitempty" yaml:"authUrl,omitempty"`
+	Cloud             string `json:"tectonic_openstack_credentials_cloud,omitempty" yaml:"cloud,omitempty"`
+	Region            string `json:"tectonic_openstack_credentials_region,omitempty" yaml:"region,omitempty"`
+	UserName          string `json:"tectonic_openstack_credentials_user_name,omitempty" yaml:"userName,omitempty"`
+	UserID            string `json:"tectonic_openstack_credentials_user_id,omitempty" yaml:"userId,omitempty"`
+	TenantID          string `json:"tectonic_openstack_credentials_tenant_id,omitempty" yaml:"tenantId,omitempty"`
+	TenantName        string `json:"tectonic_openstack_credentials_tenant_name,omitempty" yaml:"tenantName,omitempty"`
+	Password          string `json:"tectonic_openstack_credentials_password,omitempty" yaml:"password,omitempty"`
+	Token             string `json:"tectonic_openstack_credentials_token,omitempty" yaml:"token,omitempty"`
+	UserDomainName    string `json:"tectonic_openstack_credentials_user_domain_name,omitempty" yaml:"userDomainName,omitempty"`
+	UserDomainID      string `json:"tectonic_openstack_credentials_user_domain_id,omitempty" yaml:"userDomainId,omitempty"`
+	ProjectDomainName string `json:"tectonic_openstack_credentials_project_domain_name,omitempty" yaml:"projectDomainName,omitempty"`
+	ProjectDomainID   string `json:"tectonic_openstack_credentials_project_domain_id,omitempty" yaml:"projectDomainId,omitempty"`
+	DomainID          string `json:"tectonic_openstack_credentials_domain_id,omitempty" yaml:"domainId,omitempty"`
+	DomainName        string `json:"tectonic_openstack_credentials_domain_name,omitempty" yaml:"domainName,omitempty"`
+	Insecure          bool   `json:"tectonic_openstack_credentials_insecure,omitempty" yaml:"insecure,omitempty"`
+	CacertFile        string `json:"tectonic_openstack_credentials_cacert_file,omitempty" yaml:"cacert_file,omitempty"`
```

Comment:
Ah, this (and possibly other Go properties here) are currently unused.  I'll go through and cull...

Reply:
> Ah, this (and possibly other Go properties here) are currently unused. I'll go through and cull...

I've pushed a fixup to 6ba7d86362, which also alphabetizes properties in `Credentials`.  How does that look?

Reply:
looks good.  I'll apply that, rebase on master, and update this PR in a moment